tests/unit/network.test.ts

```diff
@@ -99,14 +99,12 @@ describe("Network", () => {
       await expect(network.addNode("4")).rejects.toThrow("No more addresses available in 192.168.0.0/30");
     });
 
-    it("should add a node if the network was full and some node was removed", async () => {
+    it("should not add a node to a previously assigned address", async () => {
```

Comment:
Please use positive logic and tell the story in the description: "should throw an error when there are no free IPs available"